playbooks/openstack/sample-inventory/inventory.py

```diff
@@ -98,8 +98,45 @@ def build_inventory():
             hostvars['openshift_node_labels'] = node_labels
 
         inventory['_meta']['hostvars'][server.name] = hostvars
+
+    kuryr_vars = _get_kuryr_vars(cloud)
+    if kuryr_vars:
+        inventory['OSEv3']['vars'] = kuryr_vars
+
     return inventory
 
 
+def _get_kuryr_vars(cloud_client):
+    """Returns a dictionary of Kuryr variables resulting of heat stacking"""
+    # TODO: Filter the cluster stack with tags once it is supported in shade
+    hardcoded_cluster_name = 'openshift.example.com'
+    stack = cloud_client.get_stack(hardcoded_cluster_name)
+    if stack is None or stack['stack_status'] != 'CREATE_COMPLETE':
```

Comment:
I'm not sure we want a dynamic inventory to be stack-state-aware. Let's just note instead one shall use it after the stack is created perhaps?

Reply:
IIUC how the dynamic inventory works, everytime that ansible needs inventory data, the script gets run. I am not sure there is a way to prevent the dynamic inventory from running before the stacking.

Reply:
Folks, could we please please do not introduce the finite machine for heat stacks into a synamic inventory?